infrastructure/webhook_stack/lambda.tf

```diff
@@ -0,0 +1,75 @@
+locals {
+  lambda_name = "content-prismic-webhook"
+}
+
+module "webhook_lambda" {
+  source = "git@github.com:wellcomecollection/terraform-aws-lambda?ref=v1.2.0"
+
+  name    = local.lambda_name
+  runtime = "nodejs18.x"
+  handler = "lambda.handler"
+
+  filename    = data.archive_file.empty_zip.output_path
+  memory_size = 128
+  timeout     = 30
+
+  error_alarm_topic_arn = var.lambda_alarm_topic_arn
+
+  environment = {
+    variables = {
+      SECRET_NAME    = local.webhook_secret_name
+      EVENT_BUS_NAME = aws_cloudwatch_event_bus.document_unpublish_events.name
+      EVENT_TRIGGER  = "document-unpublish"
+    }
+  }
+}
+
+resource "aws_lambda_function_url" "webhook" {
+  function_name      = module.webhook_lambda.lambda.function_name
+  authorization_type = "NONE"
+}
+
+data "archive_file" "empty_zip" {
+  output_path = "data/empty.zip"
+  type        = "zip"
+  source {
+    content  = "// This file is intentionally left empty"
+    filename = "lambda.js"
+  }
+}
+
+data "aws_iam_policy_document" "secrets_access" {
+  statement {
+    effect    = "Allow"
+    actions   = ["secretsmanager:GetSecretValue"]
+    resources = module.secrets.arns
+  }
+}
+
+data "aws_iam_policy_document" "events_publish" {
+  statement {
+    effect    = "Allow"
+    actions   = ["events:PutEvents"]
+    resources = [aws_cloudwatch_event_bus.document_unpublish_events.arn]
+  }
+}
+
+resource "aws_iam_policy" "secrets_access" {
+  name   = "${local.lambda_name}-secrets-access"
+  policy = data.aws_iam_policy_document.secrets_access.json
+}
+
+resource "aws_iam_policy" "events_publish" {
+  name   = "${local.lambda_name}-events-publish"
+  policy = data.aws_iam_policy_document.events_publish.json
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_secrets_access" {
+  role       = module.webhook_lambda.lambda_role.name
+  policy_arn = aws_iam_policy.secrets_access.arn
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_events_publish" {
+  role       = module.webhook_lambda.lambda_role.name
+  policy_arn = aws_iam_policy.events_publish.arn
+}
```

Comment:
I'm used to seeing `aws_iam_role_policy` in our codebase, something like:

```hcl
resource "aws_iam_role_policy" "secrets_access" {
  name   = module.webhook_lambda.lambda_role.name
  policy = data.aws_iam_policy_document.secrets_access.json
}
```

Is there something different about splitting it out as two separate resources, or is it just a style convention?

(I'm happy to leave it as-is even if it is just a style choice, just want to make sure I'm not missing anything.)

Reply:
The short answer is because that's how it's done elsewhere in this repo; I marginally prefer it because it completely decouples the policies from the roles but there isn't much in it

Reply:
Cool, as long as I'm not missing a significant difference.